### PR TITLE
Manual Rewrite

### PR DESCRIPTION
**Pull request by voidsong-dragonfly**
This PR does a full rewrite of sections of the manual for both clearness and organization.
Minor changes are discussed in the discord, but major ones are below:
First, categories got reorganized to better let people find items. The new list of categories is below.
 - Resources
 - Storage & Transport
 - Workbenches & Machinery
 - Construction
 - Explosives & Weaponry
 - Tools
 - Electrical Grids
 - Multiblock Constructions
 - Heavy Machinery

Multiple new pages have been added, including for minecarts, gunpowder barrels, and hempcrete.
Several pages have been factored out of others for better searchability, including strip curtains, feedthrough insulators, posts, transformers, and more.

### CODE REVIEW

File: src/datagen/java/blusunrize/immersiveengineering/data/recipes/DecorationRecipes.java [hunk 1]
```diff
@@ -192,11 +192,18 @@ private void stoneDecoration(RecipeOutput out)
...
+			.pattern("tbt")
+			.pattern("scs")
+			.define('s', Tags.Items.SAND)
+			.define('c', IETags.clay)
+			.define('t', Ingredients.HEMP_FABRIC)
+			.define('b', new IngredientFluidStack(FluidTags.WATER, FluidType.BUCKET_VOLUME))
+			.unlockedBy("has_clay", has(IETags.clay))
+			.save(new WrappingRecipeOutput<ShapedRecipe>(
+				out, r -> new TurnAndCopyRecipe(r, List.of()).allowQuarterTurn()
+			), toRL("hempcrete"));
```
Comment (BluSunrize) [inline src/datagen/java/blusunrize/immersiveengineering/data/recipes/DecorationRecipes.java]: Why is this in here, this doesn't seem to have anything to do with the manual?

File: src/main/resources/assets/immersiveengineering/lang/en_us.json [hunk 2]
```diff
@@ -731,15 +731,19 @@
   "effect.immersiveengineering.concrete_feet": "Concrete Feet",
   "effect.immersiveengineering.flashed": "Flashed",
   "manual.immersiveengineering.manual": "Engineer's Manual",
-  "manual.immersiveengineering.update": "UPDATE NEWS!",
-  "manual.immersiveengineering.general": "Overview and Resources",
-  "manual.immersiveengineering.early_machines": "Workbenches & Furnaces",
+  "manual.immersiveengineering.updates_and_introduction": "Updates & Introduction!!",
+  "manual.immersiveengineering.introduction": "Introduction",
```
Comment (BluSunrize) [inline src/main/resources/assets/immersiveengineering/lang/en_us.json:735]: Why is there two categories for Indtroduction?
"Update News" should stay as they are please, as they are exclusively auto-generated.

File: src/main/resources/assets/immersiveengineering/lang/en_us.json [hunk 3]
```diff
@@ -731,15 +731,19 @@
...
-  "manual.immersiveengineering.test": "A subcategory!?",
+  "manual.immersiveengineering.electrical_grids": "Electrical Grids",
+  "manual.immersiveengineering.explosives_weaponry": "Explosives & Weaponry",
   "manual.immersiveengineering.tools": "Tools",
-  "manual.immersiveengineering.machines": "Simple Machines",
-  "manual.immersiveengineering.heavymachines": "Heavy Machinery",
+  "manual.immersiveengineering.workbenches_machinery": "Workbenches & Machinery",
+  "manual.immersiveengineering.multiblock_constructions": "Multiblock Constructions",
+  "manual.immersiveengineering.heavy_machinery": "Heavy Machinery",
+  "manual.immersiveengineering.test": "A subcategory!?",
```
Comment (BluSunrize) [inline src/main/resources/assets/immersiveengineering/lang/en_us.json]: ...Why do we have this in here still? xD

File: src/main/resources/assets/immersiveengineering/manual/en_us/heavy_plated_shield.txt [hunk 4]
```diff
@@ -1,5 +1,5 @@
 Heavy Plated Shield
-One Man Phalanx
+One Woman Phalanx
```
Comment (BluSunrize) [inline src/main/resources/assets/immersiveengineering/manual/en_us/heavy_plated_shield.txt:2]: A bit sneaky, but I'll let it slide =P

File: src/main/resources/assets/immersiveengineering/manual/en_us/kinetic_generators.txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
+Kinetic Generators
+
```
Comment (BluSunrize) [inline src/main/resources/assets/immersiveengineering/manual/en_us/kinetic_generators.txt]: Oh this never got a subtitle, huh...?

File: src/main/resources/assets/immersiveengineering/manual/en_us/storage_minecarts.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+Storage Minecarts
+Let The Bullgine Run
```
Comment (BluSunrize) [inline src/main/resources/assets/immersiveengineering/manual/en_us/storage_minecarts.txt:2]: Can't help but notice that you overrode me in the end ;)

**Review by BluSunrize — CHANGES_REQUESTED** [review]
The recipe change and "Update" category should be addressed.

File: src/main/resources/assets/immersiveengineering/manual/en_us/storage_minecarts.txt (hunk 6 above)
Comment (voidsong-dragonfly) [reply src/main/resources/assets/immersiveengineering/manual/en_us/storage_minecarts.txt:2]: Yeah, I liked this shanty better. I can change it to the one you suggested if you want?

File: src/main/resources/assets/immersiveengineering/manual/en_us/kinetic_generators.txt (hunk 5 above)
Comment (voidsong-dragonfly) [reply src/main/resources/assets/immersiveengineering/manual/en_us/kinetic_generators.txt]: Oh, huh, didn't realize that

File: src/main/resources/assets/immersiveengineering/lang/en_us.json (hunk 3 above)
Comment (voidsong-dragonfly) [reply src/main/resources/assets/immersiveengineering/lang/en_us.json]: I 'unno, I left it. I'll remove it though.

File: src/datagen/java/blusunrize/immersiveengineering/data/recipes/DecorationRecipes.java (hunk 1 above)
Comment (voidsong-dragonfly) [reply src/datagen/java/blusunrize/immersiveengineering/data/recipes/DecorationRecipes.java]: Ooop, will remove!

**Comment by BluSunrize** [comment]
Hell Yeah

**Pull request merged**